assets/js/vue-apps/form-components/address-lookup.vue

```diff
@@ -1,254 +1,255 @@
+            } else if (this.fullAddress.postcode !== null) {
+                this.currentState = this.states.AlreadyAnswered;
+            } else {
+                this.currentState = states.NotAsked;
+            }
+        });
 
-            const $form = $(this.$el)
-                .parents('form')
-                .find('input[type="submit"]');
-            const that = this;
-            $form.on('click', function() {
-                if (
-                    that.candidates.length === 0 &&
-                    that.postcode
-                ) {
-                    // @TODO i18n
-                    alert(
-                        `Please click "Find address" and choose an address from the list.`
-                    );
-                    trackEvent(
-                        'Form warning',
-                        'Postcode lookup',
-                        'Typed but not submitted'
-                    );
-                    document.querySelector('.address-lookup').scrollIntoView();
-                    // Prevent form submission (for nested)
-                    return false;
+        if (this.address) {
+            try {
+                const addressParts = JSON.parse(this.address);
+                // Ensure a blank address doesn't trigger AlreadyAsked state
+                if (addressParts && addressParts.postcode) {
+                    this.currentAddress = addressParts;
+                    const fullAddress = {
+                        line1: this.currentAddress.line1,
+                        line2: this.currentAddress.line2,
+                        townCity: this.currentAddress.townCity,
+                        county: this.currentAddress.county,
+                        postcode: this.currentAddress.postcode
+                    };
+                    this.currentState = this.states.AlreadyAnswered;
+                    this.fullAddress = fullAddress;
                 }
-            });
-        },
-        methods: {
-            getAddressFromId(udprn) {
-                return this.addressData.find(_ => _.udprn === udprn);
-            },
-            formatAddress(address) {
-                return {
-                    line1: address.line_1,
-                    line2: address.line_2,
-                    townCity: address.post_town,
-                    county: address.county,
-                    postcode: address.postcode
-                };
-            },
-            handleFailure() {
-                // Handle case of no results for search
-                this.currentState = this.states.Failure;
+            } catch (e) {} // eslint-disable-line no-empty
+        }
+
+        const $form = $(this.$el)
+            .parents('form')
+            .find('input[type="submit"]');
+        const that = this;
+        $form.on('click', function() {
+            if (
+                that.candidates.length === 0 &&
+                that.postcode &&
+                that.currentState !== states.NotRequired
```

Comment:
This line is the only real change here (sorry, prettier/whitespace otherwise).

It fixes the scenario where someone clicks "No" here, enters a postcode, then clicks "Yes" (eg. hides the widget) and then submits the step – they were seeing the `alert()` even though the field wasn't required anymore (although their browser still took them to the next step).

![image](https://user-images.githubusercontent.com/394376/62284071-d2e03780-b44a-11e9-8e35-3c8309145ad2.png)